src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs

```diff
@@ -744,15 +776,20 @@ private void ParseInstanceMethodEntrypoints(bool[] isEntryPoint)
                 {
                     isEntryPoint[method.EntryPointRuntimeFunctionId] = true;
                 }
-                _methods.Add(method);
+                if (!Methods.TryGetValue(instMethodEntryPointSection, out List<ReadyToRunMethod> sectionMethods))
+                {
+                    sectionMethods = new List<ReadyToRunMethod>();
+                    Methods.Add(instMethodEntryPointSection, sectionMethods);
+                }
+                sectionMethods.Add(method);
                 _instanceMethods.Add(new InstanceMethod(curParser.LowHashcode, method));
                 curParser = allEntriesEnum.GetNext();
             }
         }
 
         private void CountRuntimeFunctions(bool[] isEntryPoint)
         {
-            foreach (ReadyToRunMethod method in _methods)
+            foreach (ReadyToRunMethod method in Methods.Values.SelectMany(sectionMethods => sectionMethods))
```

Comment:
Why cannot we do just 
```suggestion
            foreach (ReadyToRunMethod method in Methods.Values)
```
?

Reply:
This is because the method lists are now split by R2R sections (to let us separately dump method entrypoints for each component assembly in the composite R2R). Because of this Methods.Values is basically an <code>IEnumerable<List<ReadyToRunMethod>></code>.

Reply:
Ah, my mistake, I've somehow missed the one extra level.